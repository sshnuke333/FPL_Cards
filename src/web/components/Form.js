import React from 'react';
import { InputControl, Input, Output } from './Form.styles';

export const Form = ({ updateStatus }) => {
    return (
        <InputControl data-testid="slider" id="settings">
            <label htmlFor="deck">choose deck size:</label>
            <Input
                type="range"
                aria-label="slider"
                aria-valuemin={10}
                aria-valuemax={22}
                aria-valuenow={22}
                id="deck"
                name="deck"
                min="10"
                max="22"
                step="4"
                onInput={updateStatus}
            ></Input>
            <Output aria-label="status" name="result" htmlFor="deck">
                22 Cards
            </Output>
        </InputControl>
    );
};
