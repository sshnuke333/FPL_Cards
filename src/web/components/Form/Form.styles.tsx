import styled from 'styled-components';

export const InputControl = styled.div`
    display: none;
    padding: 1rem;
    color: white;
`;

export const Input = styled.input`
    display: block;
    padding: 0.5rem 0;
    cursor: pointer;
    &:focus {
        outline: none;
    }
`;

export const Output = styled.output`
    display: block;
`;
