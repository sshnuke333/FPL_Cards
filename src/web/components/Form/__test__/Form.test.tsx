import { render, screen } from '@testing-library/react';
import Form from '../Form';

describe('Form component', () => {
    beforeEach(() => {
        render(<Form updateStatus={() => {}} />);
    });
    it('renders correctly', () => {
        const { asFragment } = render(<Form updateStatus={() => {}} />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('renders a range input', () => {
        expect(screen.getByLabelText('slider')).toBeInTheDocument();
    });
    it('renders a output', () => {
        expect(screen.getByLabelText('status')).toBeInTheDocument();
    });
    it('renders a label for input', () => {
        expect(screen.getByText(/choose/i)).toBeInTheDocument();
    });
});
