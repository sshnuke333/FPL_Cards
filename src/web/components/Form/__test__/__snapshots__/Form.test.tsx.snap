// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe kLvQKC"
    data-testid="slider"
    id="settings"
  >
    <label
      for="deck"
    >
      choose deck size:
    </label>
    <input
      aria-label="slider"
      aria-valuemax="22"
      aria-valuemin="10"
      aria-valuenow="22"
      class="sc-gswNZR hNcShn"
      id="deck"
      max="22"
      min="10"
      name="deck"
      step="4"
      type="range"
    />
    <output
      aria-label="status"
      class="sc-dkrFOg gDZyuj"
      for="deck"
      name="result"
    >
      22 Cards
    </output>
  </div>
</DocumentFragment>
`;
